app/models/asset_field_type.rb

```diff
@@ -52,6 +53,29 @@ def mapping
 
   private
 
+  def is_image(mime_type)
+    ['image/jpeg', 'image/pjpeg', 'image/png','application/pdf' ,'image/x-png', 'image/gif'].include?(mime_type)
+  end
+
+  def format_data(metadata_hash)
+   return metadata_hash unless is_image(metadata_hash[:content_type])
+   metadata_hash.merge(image_asset_styles)
+  end
+
+  def image_asset_styles
+    {
+      styles: {
```

Comment:
Ok, then maybe we should have an array of image sizes and their dimensions that isn't set with the key `styles`, we could even change it to `image_asset_styles` which we could pass as styles to Paperclip in case it is an image. When Paperclip sees that there is a styles object, it will pass whatever the file is to ImageMagic no matter what.